test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,376 @@
+            {
+            }
+            collection &coll;
+            scoped_cursor cursor;
+        };
+
+        /* Collection cursor vector. */
+        std::vector<collection_cursor> ccv;
+        uint64_t collection_count = tc->db.get_collection_count();
+        uint64_t collections_per_thread = collection_count / tc->thread_count;
+
+        /* Must have unique collections for each thread. */
+        testutil_assert(collection_count % tc->thread_count == 0);
+        for (int i = tc->id * collections_per_thread;
+             i < (tc->id * collections_per_thread) + collections_per_thread && tc->running(); ++i) {
+            collection &coll = tc->db.get_collection(i);
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
+            ccv.push_back({coll, std::move(cursor)});
+        }
+
+        std::string key;
+        uint64_t counter = 0;
+
+        while (tc->running()) {
+
+            auto &cc = ccv[counter];
+            tc->transaction.begin();
+
+            while (tc->transaction.active() && tc->running()) {
+
+                /* Generate a random key. */
+                key = random_generator::instance().generate_random_string(tc->key_size);
+
+                /* Insert a key value pair. */
+                if (tc->insert(cc.cursor, cc.coll.id, key)) {
+                    if (tc->transaction.can_commit())
+                        /* We are not checking the result of commit as it is not necessary. */
+                        tc->transaction.commit();
+                } else {
+                    tc->transaction.rollback();
+                }
+
+                /* Sleep the duration defined by the configuration. */
+                tc->sleep();
+            }
+
+            /* Rollback any transaction that could not commit before the end of the test. */
+            if (tc->transaction.active())
+                tc->transaction.rollback();
+
+            /* Reset our cursor to avoid pinning content. */
+            testutil_check(cc.cursor->reset(cc.cursor.get()));
+            if (++counter == collections_per_thread)
+                counter = 0;
+            testutil_assert(counter < collections_per_thread);
+        }
+    }
+
+    void
+    read_operation(test_harness::thread_context *tc) override final
+    {
+        /*
+         * Each read operation performs search_near calls with and without prefix enabled on random
+         * collections. Each prefix is randomly generated. The result of the seach_near call with
+         * prefix enabled is then validated using the search_near call without prefix enabled.
+         */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        std::map<uint64_t, scoped_cursor> cursors;
+        // TODO - Should this be a fixed value ? Should it be from the configuration ?
+        const uint64_t prefix_size = 3;
```

Comment:
Yeah I think it is somewhat dependent on the key_size() and how many inserts threads are performing at once, such that searching for a long prefix size would be useful. Since searching for a long prefix size, will limit the number of entries that are visible to the entry.